Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/FilterScreen.kt

```diff
@@ -0,0 +1,276 @@
+import com.google.accompanist.flowlayout.FlowMainAxisAlignment
+import com.google.accompanist.flowlayout.FlowRow
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun FilterScreen(
+    onDismiss: () -> Unit
+) {
+    var sortChanged by remember { mutableStateOf(false) }
+    var sortState by remember { mutableStateOf(SnackRepo.getSortDefault()) }
+    var maxCalories by remember { mutableStateOf(0f) }
+
+    Dialog(onDismissRequest = onDismiss) {
+
+        val priceFilters = remember { SnackRepo.getPriceFilters() }
+        val categoryFilters = remember { SnackRepo.getCategoryFilters() }
+        val lifeStyleFilters = remember { SnackRepo.getLifeStyleFilters() }
+        JetsnackScaffold(
+            topBar = {
+                TopAppBar(
+                    navigationIcon = {
+                        IconButton(onClick = onDismiss) {
+                            Icon(
+                                imageVector = Icons.Filled.Close,
+                                contentDescription = stringResource(id = R.string.close)
+                            )
+                        }
+                    },
+                    title = {
+                        Text(
+                            text = stringResource(id = R.string.label_filters),
+                            modifier = Modifier.fillMaxWidth(),
+                            textAlign = TextAlign.Center,
+                            style = MaterialTheme.typography.h6
+                        )
+                    },
+                    actions = {
+                        var resetEnabled = sortChanged
+                        IconButton(
+                            onClick = { /* TODO: Open search */ },
+                            enabled = resetEnabled
+                        ) {
+                            val alpha = if (resetEnabled) {
+                                ContentAlpha.high
+                            } else {
+                                ContentAlpha.disabled
+                            }
+                            CompositionLocalProvider(LocalContentAlpha provides alpha) {
+                                Text(
+                                    text = stringResource(id = R.string.reset),
+                                    style = MaterialTheme.typography.body2
+                                )
+                            }
+                        }
+                    },
+                    backgroundColor = JetsnackTheme.colors.uiBackground
+                )
+            }
+        ) {
+            Column(
+                Modifier
+                    .fillMaxSize()
+                    .verticalScroll(rememberScrollState())
+                    .padding(horizontal = 24.dp, vertical = 16.dp),
+            ) {
+                SortFiltersSection(
+                    sortState = sortState,
+                    onFilterChange = { filterName ->
+                        sortChanged = filterName != SnackRepo.getSortDefault()
```

Comment:
Perhaps hold the default sort in a val rather than querying it in every callback?

Reply:
Done